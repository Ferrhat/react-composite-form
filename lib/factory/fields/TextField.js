import React, {Component} from "react";
import { FormControl } from 'react-bootstrap';
import { defaultTo, get } from 'lodash';

class TextField extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        const { name, label, value, valueUpdateHandler, changeHandler, disabled } = this.props;

        const onChange = (evt) => {
            valueUpdateHandler(name, evt.target.value);
            changeHandler(evt);
        }

        return (
            <div>
                <FormControl
                    value={ defaultTo(value, '') }
                    onChange={ onChange }
                    type="text"
                    componentClass={ 'input' }
                    placeholder={ label }
                    disabled={ disabled }
                />
                <div className={'validationErrorMessage'}>{get(this.props.errors, '0.message', '')}</div>
            </div>
        );
    }
}

export default TextField;
