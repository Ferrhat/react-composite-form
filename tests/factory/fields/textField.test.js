import React from 'react';
import {shallow, mount} from 'enzyme';
import TextField from "../../../lib/factory/fields/TextField";


describe('TextField', () => {
    const mockValueUpdateHandler = jest.fn();
    const mockChangeHandler = jest.fn();
    const textField = shallow(<TextField valueUpdateHandler={mockValueUpdateHandler} changeHandler={mockChangeHandler} name={'testName'} />);

    it('renders properly', () => {
        expect(textField).toMatchSnapshot();
    });

    it('contains a connected FormControl component', () => {
        expect(textField.find('FormControl').length).toEqual(1);
    });

    it('should trigger a change event', () => {
        const event = {
            preventDefault() {},
            target: { name: 'testName', value: 'testValue' }
        };
        textField.find('FormControl').simulate('change', event);
        expect(mockValueUpdateHandler).toBeCalledWith('testName', 'testValue');
        expect(mockChangeHandler).toBeCalledWith(event);
    });

});
