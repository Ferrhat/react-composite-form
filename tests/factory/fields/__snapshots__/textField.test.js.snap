// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders properly 1`] = `
<div>
  <FormControl
    bsClass="form-control"
    componentClass="input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="validationErrorMessage"
  />
</div>
`;
